src/System.Buffers/src/System/Buffers/NativeBufferBucket.cs

```diff
@@ -33,13 +34,26 @@ internal NativeBufferBucket(int elementsInBuffer, int numberOfBuffers)
             }
         }
 
-        ~NativeBufferBucket()
```

Comment:
So the Finalizer should check if we need to dispose and call Dispose if necessary?